bids/layout/tests/test_path_building.py

```diff
@@ -22,12 +22,12 @@ def test_invalid_file_construction(layout):
     ents = dict(subject='01', run=1, task='resting-state', suffix='bold')
     with pytest.raises(ValueError):
         layout.build_path(ents)
-    
+
     target = "sub-01/func/sub-01_task-resting-state_run-1_bold.nii.gz"
     assert layout.build_path(ents, validate=False) == target
 
 
 def test_failed_file_construction(layout):
     ents = dict(subject='01', fakekey='foobar')
     with pytest.raises(ValueError):
-        layout.build_path(ents)
+        layout.build_path(ents, strict=True)
```

Comment:
Could you add:

```Python
@pytest.mark.parametrize("strict", [True, False])
@pytest.mark.parametrize("validate", [True, False])
def test_insufficient_entities(layout, strict, validate):
    with pytest.raises(ValueError):
        layout.build_path({'subject': '01'}, strict=strict, validate=validate)
```

Reply:
This might fall a bit outside the scope in the sense that the test is over `BIDSLayout.build_path` not over the actual function being modified. My worry is that, since I haven't been looking into `BIDSLayout.build_path` in-depth, the ValueError might be raised for other reasons too. Do you mind if I split this suggestion to its own PR?

(sorry, your review and my last comment happened simultaneously)

Reply:
That's fine.